### PR TITLE
Fix custom tracing b58 printing

### PR DESCRIPTION
**Pull request by macpie**

### CODE REVIEW

File: custom_tracing/Cargo.toml [hunk 1]
```diff
@@ -6,22 +6,23 @@ license.workspace = true
 edition.workspace = true
 
 [dependencies]
-serde = { version = "1", features = ["derive"] }
-notify = { version = "6", default-features = false }
 anyhow = "1"
+axum = { version = "0.7", features = ["tracing"], optional = true }
+bs58 = { version = "0.4", features = ["check"] }
```
Comment (madninja) [inline custom_tracing/Cargo.toml:11]: don'tt be this precise in a lib this far upstream? Make version = "0" or version = ">=..." if it needs a minimum version

File: custom_tracing/src/lib.rs [hunk 2]
```diff
@@ -54,7 +55,7 @@ where
 }
 
 pub fn record_b58(key: &str, pub_key: &[u8]) {
-    let b58 = bs58::encode(pub_key).into_string();
+    let b58 = PublicKeyBinary::from(pub_key).to_string();
```
Comment (madninja) [inline custom_tracing/src/lib.rs:58]: huh? what format was that key string thing there?

Comment (macpie) [reply custom_tracing/src/lib.rs:58]: Just wanted to print b58 my old stuff was bad did not realized that `to_string` on a `PublicKeyBinary` or `PublicKey` gives you the b58 string version

**Review by bbalser — APPROVED** [review]

**Review by michaeldjeffrey — APPROVED** [review]

**Comment by michaeldjeffrey** [comment]
Can we get a description of why this was the wrong way to print a b58?

**Comment by madninja** [comment]
> Can we get a description of why this was the wrong way to print a b58?

in helium crypto we use b58 *check encoding* (with a version 0 prefix) not a plain b58 encoding

**Pull request merged**